### DISCUSSION TITLE
Appium 2.0 installation fails

### DISCUSSION THREAD

**Discussion by kdave47** [appium/appium#18422] [Q&A]
appium -v

Fatal Error: Appium had trouble loading the extension installation cache file (/usr/local/bin/appium/node_modules/.cache/appium/extensions.yaml). It may be invalid YAML. Specific error: ENOTDIR: not a directory, open ‘/usr/local/bin/appium/node_modules/.cache/appium/extensions.yaml’
at /usr/local/lib/node_modules/appium/lib/extension/manifest.js:410:19
at Manifest.read (/usr/local/lib/node_modules/appium/lib/extension/manifest.js:463:7)
at loadExtensions (/usr/local/lib/node_modules/appium/lib/extension/index.js:22:3)
at init (/usr/local/lib/node_modules/appium/lib/main.js:175:40)
at main (/usr/local/lib/node_modules/appium/lib/main.js:283:22)

Unable to install appium using npm on mac.

**Comment by jlipps** [comment]
This should probably be an issue and not a discussions. do you have APPIUM_HOME set?

**Reply by kdave47** [reply]
Yes, I do have it set 
<img width="439" alt="image" src="https://user-images.githubusercontent.com/4139979/229416375-a22cfef0-e004-4e8e-820a-dbc79f963b63.png">


Also I have working appium with previous version:
<img width="301" alt="image" src="https://user-images.githubusercontent.com/4139979/229416552-d1bc0020-aba6-4de0-8db5-3a35a9616b32.png">

**Reply by jlipps** [reply]
ok, try to `rm -rf $APPIUM_HOME` and then reinstall appium. and make sure you `npm uninstall -g appium` before installing appium 2

**Comment by kdave47** [comment]
Thank you for your response, I did same steps as you suggested and was able to install appium 2.0 version. What i observed is once I am able to get --version If i set the APPIUM_HOME  again in zshrc it is causing issue again.

Steps I followed:

1. Uninstalled Appium 
2. Cleared npm cache 
3. Removed APPIUM_HOME from zshrc
4. Installed next version of appium
5. Checked version [Successful installation]
6.  Set APPIUM_HOME  again in  .zshrc
7. Checked version  [THROWS Error as below screenshot]


Note: Removing APPIUM_HOME from zshrc works and shows successful installed APPIUM 2.0.

Kindly advice if anything wrong or unexpected step in above walkthrough.

<img width="1266" alt="image" src="https://user-images.githubusercontent.com/4139979/229985441-a0ab34af-fec7-47bd-a7d8-2273e20323ae.png">

**Reply by jlipps** [reply]
why do want APPIUM_HOME to be `/usr/local/bin/appium`? Try to `rm -rf /usr/local/bin/appium` and then let Appium recreate its folder structure there. Seems like there may be a file ownership problem on your system somehow.

**Comment by kdave47** [comment]
Following as above guidance worked, Thanks for your help @jlipps  Closing this Q&A.
